### DISCUSSION TITLE
auto_lr_find gives error (NameError: name 'trainer' is not defined)

### DISCUSSION THREAD

**Discussion by lyndonlauder** [Lightning-AI/pytorch-lightning#13330] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi i am trying to use lightning auto_lr_find, i use the command trainer.tune(model) and it gives me this error 

NameError: name 'trainer' is not defined

Here is the model code and link

https://github.com/kuielab/mdx-net/blob/main/src/models/mdxnet.py
https://github.com/kuielab/mdx-net/blob/8e0476f29f330e2fe31cbb2833737ad0b2df7ad0/src/utils/utils.py

Please do you have any advice for this?

**Comment by akihironitta** [comment]
@lyndonlauder Could you please provide your env detail and full error message? You can use this template here: https://github.com/Lightning-AI/lightning/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Reply by akihironitta** [reply]
@lyndonlauder I've converted your issue to a GitHub Discussion because this is the right place to ask questions and GitHub issues are basically used for bug reports or feature requests!

---

As the error states, there seems no trainer defined before you run `trainer.fit(...)` :)

**Reply by lyndonlauder** [reply]
@akihironitta do you know how i can define trainer because i am not experienced with code. Thank you for your time.

**Reply by akihironitta** [reply]
> ```
> NameError Traceback (most recent call last)
> in ()
> ----> 1 trainer.fit(model)
> 
> NameError: name 'trainer' is not defined
> ```

Where is the `trainer.fit` call exactly? I couldn't find any from the links you've provided above.

**Reply by lyndonlauder** [reply]
@akihironitta here https://github.com/kuielab/mdx-net/blob/002b33f2b4c0dbc8bb2b93ac880fafde2d42ff6f/src/train.py#L88

**Reply by akihironitta** [reply]
It should be defined at: https://github.com/kuielab/mdx-net/blob/002b33f2b4c0dbc8bb2b93ac880fafde2d42ff6f/src/train.py#L71

Unfortunately, I'm not familiar with `hydra`. ~I'd suggest you creating an issue in their repository.~ You have already :) https://github.com/kuielab/mdx-net/issues/37

**Comment by lyndonlauder** [comment]
Thank you @akihironitta here it what you asked for

To Reproduce: trainer.tune(model)
Expected behavior: auto_lr_find works
Environment: conda google colab
PyTorch Lightning Version (e.g., 1.5.0): 1.6.4
PyTorch Version (e.g., 1.10): 1.8.1
Python version (e.g., 3.9): 3.8
OS (e.g., Linux): Google colab
CUDA/cuDNN version: cudatoolkit-11.7.0 
GPU models and configuration: google colab p100
How you installed PyTorch (conda, pip, source): conda

Full error:

NameError                                 Traceback (most recent call last)
<ipython-input-38-45d4afebefac> in <module>()
----> 1 trainer.fit(model)

NameError: name 'trainer' is not defined
